contracts/BandOracle.cdc

```diff
@@ -7,37 +7,36 @@ pub contract BandOracle {
     /// Paths
     ///
 
-    // Admin resource storage path
-    pub let BandOracleAdminStoragePath: StoragePath
+    // OracleAdmin resource paths
+    pub let OracleAdminStoragePath: StoragePath
+    pub let OracleAdminPrivatePath: PrivatePath
+
+    // Relay resource paths
+    pub let RelayStoragePath: StoragePath
+    pub let RelayPrivatePath: PrivatePath
 
     ///
-    /// Contract level fields
+    /// Fields
     ///
     
+    /// Set a string as base private path for data updater capabilities
+    access(contract) let dataUpdaterPrivateBasePath: String
+
     // Mapping from symbol to data struct
     access(contract) let symbolsRefData: {String: RefData}
 
     ///
     /// Events
     ///
-
-    //
-    pub event NewRelayerCreated()
-    
-    //
-    pub event RelayerAuthorised()
-    
-    //
-    pub event RelayerDismissed()
     
-    //
-    pub event RefDataUpdated()
+    // Emitted by a relayer when it updates a set of symbols
+    pub event RefDataUpdated(symbols: [String], relayerID: UInt64, requestID: UInt64)
```

Comment:
Actually I included relayerID very intentionally there, as a safeguard for Band to keep track of which relayer has done which update, in case one of them does a malicious update. Thoughts?